docs/reference/search/search-your-data/retrievers-reranking/semantic-reranking.asciidoc

```diff
@@ -0,0 +1,153 @@
+* There are significantly fewer open-source cross-encoders, while bi-encoders offer a wide variety of sizes, languages, and other trade-offs.
+* The effectiveness of cross-encoders can also improve the relevance of semantic retrievers.
+For example, their ability to take word order into account can improve on dense or sparse embedding retrieval.
+* When trained in tandem with specific retrievers (like lexical/BM25), cross-encoders can “correct” typical errors made by those retrievers.
+* Cross-encoders output scores that are consistent across queries.
+This means enables you to maintain high relevance in result sets, by setting a minimum score threshold for all queries.
+For example, this is important when using results in a RAG workflow or if you're otherwise feeding results to LLMs.
+Note that similarity scores from bi-encoders/embedding similarities are _query-dependent_, meaning you cannot set universal cut-offs.
+* Bi-encoders rerank using embeddings. You can improve your reranking latency by creating embeddings at ingest-time. These embeddings can be stored for reranking without being indexed for retrieval, reducing your memory footprint.
+==============
+
+[discrete]
+[[semantic-reranking-in-es]]
+==== Semantic reranking in {es}
+
+In {es}, semantic rerankers are implemented using the {es} *Inference API* and a <<retriever,retriever>>.
+
+To use semantic reranking in {es}, you need to:
+
+. Choose a reranking model. In addition to cross-encoder models running on {es} inference nodes, we also expose external models and services via the Inference API to semantic rerankers.
+** This includes cross-encoder models running in https://huggingface.co/inference-endpoints[HuggingFace Inference Endpoints] and the https://cohere.com/rerank[Cohere Rerank API].
+. Create a `rerank` task using the <<put-inference-api,{es} Inference API>>.
+The Inference API creates an inference endpoint and configures your chosen machine learning model to perform the reranking task.
+. Define a `text_similarity_reranker` retriever in your search request.
+The retriever syntax makes it simple to configure both the retrieval and reranking of search results in a single API call.
+
+.*Example search request* with semantic reranker
+[%collapsible]
+==============
+The following example shows a search request that uses a semantic reranker to reorder the top-k documents based on their semantic similarity to the query.
+[source,console]
+----
+POST _search
+{
+  "retriever": {
+    "text_similarity_reranker": {
+      "retriever": {
+        "standard": {
+          "query": {
+            "match": {
+              "text": "How often does the moon hide the sun?"
+            }
+          }
+        }
+      },
+      "field": "text",
+      "inference_id": "my-cohere-rerank-model-v3",
+      "inference_text": "How often does the moon hide the sun?",
+      "rank_window_size": 100,
+      "min_score": 0.5
+    }
+  },
+  "_source": ["title", "text"],
+  "size": 10
```

Comment:
These properties are not related to reranking, I suggest removing them in order to simplify the example.
```suggestion
  }
```